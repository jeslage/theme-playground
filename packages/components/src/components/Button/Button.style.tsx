import styled from 'styled-components';

const StyledButton = styled.button`
  outline: none;
  cursor: pointer;
  margin: 0.5rem 0.25rem;
  border: 0;
  border-radius: 3em;
  padding: 0.75rem 1rem;
  text-align: center;
  transition: all 150ms ease-out;
  font-size: 13px;
  font-weight: 700;
  line-height: 1;
  color: ${({ theme }) => theme.text};
  background: transparent;
  border: 1px solid ${({ theme }) => theme.text};

  &:hover {
    background: ${({ theme }) => theme.border};
    border-color: ${({ theme }) => theme.border};
    color: ${({ theme }) => theme.backgroundDark};
  }

  &:active {
    color: ${({ theme }) => theme.backgroundDark};
    background: ${({ theme }) => theme.text};
    color: ${({ theme }) => theme.backgroundDark};
  }

  svg {
    display: inline-block;
    height: 16px;
    width: 16px;
    vertical-align: top;
    margin-right: 6px;
    margin-top: -1px;
    margin-bottom: -1px;
    pointer-events: none;
  }
`;

export default StyledButton;
