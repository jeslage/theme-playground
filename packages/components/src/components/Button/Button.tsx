import * as React from 'react';

import StyledButton from './Button.style';

export type ButtonProps = { onClick: () => void };

const Button: React.FC<ButtonProps> = ({ children, onClick }) => (
  <StyledButton type="button" onClick={onClick}>
    {children}
  </StyledButton>
);

export default Button;
