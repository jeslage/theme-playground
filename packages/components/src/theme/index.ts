export const darkTheme = {
  base: 'dark',
  backgroundDark: '#2f2f2f',
  backgroundLight: '#333',
  border: '#999999',
  text: '#FFFFFF'
};

export const lightTheme = {
  base: 'light',
  backgroundDark: '#f6f9fc',
  backgroundLight: '#fff',
  border: '#666666',
  text: '#333333'
};

export const getUiTheme = theme => {
  if (theme === undefined) return darkTheme;

  if (typeof theme === 'string') {
    if (theme === 'dark') {
      return darkTheme;
    } else if (theme === 'light') {
      return lightTheme;
    }

    throw Error(
      `Theme "${theme}" is not available. Please use "light" or "dark", or provide a custom theme object.`
    );
  }

  if (theme.base && theme.base === 'light') {
    return {
      ...lightTheme,
      ...theme
    };
  }

  return {
    ...darkTheme,
    ...theme
  };
};
