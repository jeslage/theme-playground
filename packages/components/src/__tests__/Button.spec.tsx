import 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';

import Button from '../components/Button/Button';
import StyledButton from '../components/Button/Button.style';
import { darkTheme } from '../theme';

describe('Button', () => {
  it('Button should render correctly', () => {
    const component = shallow(
      <Button onClick={() => console.log('Test')}>Test</Button>
    );

    expect(component).toMatchSnapshot();
  });

  it('StyledButton should render correctly', () => {
    const component = shallow(<StyledButton theme={{ colors: darkTheme }} />);

    expect(component).toMatchSnapshot();
  });
});
