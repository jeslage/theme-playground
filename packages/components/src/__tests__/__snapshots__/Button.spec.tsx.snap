// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button should render correctly 1`] = `
<styled.button
  onClick={[Function]}
  type="button"
>
  Test
</styled.button>
`;

exports[`Button StyledButton should render correctly 1`] = `
<button
  className="sc-AxjAm LDa-Di"
/>
`;
